pygmt/src/grdtrack.py

```diff
@@ -283,26 +287,65 @@ def grdtrack(points, grid, newcolname=None, outfile=None, **kwargs):
     ...     points=points, grid=grid, newcolname="bathymetry"
     ... )
     """
+    # pylint: disable=too-many-branches
     if hasattr(points, "columns") and newcolname is None:
         raise GMTInvalidInput("Please pass in a str to 'newcolname'")
 
+    if points is None and kwargs.get("E") is None:
+        raise GMTInvalidInput("Must give 'points' or set 'profile'.")
+
+    if points is not None and kwargs.get("E") is not None:
+        raise GMTInvalidInput("Can't set both 'points' and 'profile'.")
+
+    # Backward compatibility with old parameter order "points, grid".
+    # deprecated_version="0.7.0", remove_version="v0.9.0"
+    is_a_grid = True
+    if not isinstance(grid, (xr.DataArray, xr.Dataset, str)):
+        is_a_grid = False
```

Comment:
Don't think `xr.Dataset` is treated as a grid?
```suggestion
    if not isinstance(grid, (xr.DataArray, str)):
        is_a_grid = False
```